projects/components/src/action-menu/action-menu.component.ts

```diff
@@ -23,6 +30,21 @@ export function getDefaultActionDisplayConfig(cfg: ActionDisplayConfig = {}): Ac
     return { ...defaults, ...cfg };
 }
 
+/**
+ * We internally convert the callbacks to booleans to avoid calling the callbacks all the time from template. However, we don't want to
+ * allow callers to assign boolean variables to availability as there is no way to know when those variables can get updated from outside.
+ */
+interface ActionItemInternal<R, T> extends BaseActionItem<R, T> {
+    /**
+     * Used for determining where in the action menu this action gets displayed
+     */
+    actionType?: ActionType;
+    /**
+     * Condition whether or not the action is available.
+     */
+    availability?: ((selectedEntities: R[]) => boolean) | Observable<boolean> | boolean;
```

Comment:
I thought we decided that `((selectedEntities: R[]) => boolean)` shouldn't be in the internal availability?

To get around the children problem, you might want to try redeclaring `children` in this interface with the type `ActionItemInternal`.

Reply:
>To get around the children problem, you might want to try redeclaring children in this interface with the type ActionItemInternal.
That is not working as TS complains that action types are incompatible

Reply:
I still wanna push back on this. We shouldn't be modifying the definition of `availability` to get some Typescript issue to go away. Having the callback still in this type is confusing and could lead to problems down the line. 

Maybe children need to be given a type of `BaseActionItem`? And then in here they're redeclared to be `ActionItemInternal`? Like, there's some way to rework these types so they pass without any casts or hacks.

Reply:
I removed the call back from ActionItemInternal. However, I still required the casts in 2 places. I think that is fine because, at those points in the code, the action children are converted into ActionItemInternal but TS doesn't know that and we don't want to expose children as ActionItemInternal.